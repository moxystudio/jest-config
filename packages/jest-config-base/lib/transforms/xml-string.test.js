'use strict';

const fs = require('fs');
const tmp = require('tmp-sync');

const dir = tmp.in(__dirname);

afterAll(() => {
    tmp.clean();
});

it('should return XML string', () => {
    const contents = `
<?xml version="1.0" encoding="UTF-8"?>
<todos>
    <todo id="1">
        <description>foo</description>
        <done>true</done>
    </todo>
    <todo id="2">
        <description>bar</description>
        <done>false</done>
    </todo>
</todos>
`;
    const filename = `${dir}/test-file.xml`;

    fs.writeFileSync(filename, contents);

    expect(require(filename)).toMatchSnapshot();
});
