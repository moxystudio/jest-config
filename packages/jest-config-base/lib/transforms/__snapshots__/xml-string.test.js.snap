// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return XML string 1`] = `
"
<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<todos>
    <todo id=\\"1\\">
        <description>foo</description>
        <done>true</done>
    </todo>
    <todo id=\\"2\\">
        <description>bar</description>
        <done>false</done>
    </todo>
</todos>
"
`;
